### PR TITLE
Confirm your Training Provider

### PR DESCRIPTION
**Pull request by benagain**

### CODE REVIEW

File: src/SFA.DAS.ApprenticeCommitments.Database/Tables/Apprenticeship.sql [hunk 1]
```diff
@@ -3,7 +3,10 @@
 	[Id] BIGINT IDENTITY(10000,1) NOT NULL,
 	[ApprenticeId] UNIQUEIDENTIFIER NOT NULL, 
     [CommitmentsApprenticeshipId] BIGINT NOT NULL,
-	CONSTRAINT PK_Apprenticeship_Id PRIMARY KEY CLUSTERED ([Id]),
+	[EmployerName] NVARCHAR(100) NOT NULL, 
+	[EmployerAccountLegalEntityId] BIGINT NOT NULL,
+    [TrainingProviderName] NVARCHAR(100) NULL,
```
Comment (Pauljgraham) [inline src/SFA.DAS.ApprenticeCommitments.Database/Tables/Apprenticeship.sql:8]: Training provider Name should not be nullable, Also thinking about it. Should we also be saving the ProviderId as this uniquely identifies the provider (same logic as the employer)

File: src/SFA.DAS.ApprenticeCommitments.Database/Tables/Registration.sql [hunk 2]
```diff
@@ -6,5 +6,8 @@
     [Email] NVARCHAR(150) NOT NULL, 
     [UserIdentityId] UNIQUEIDENTIFIER NULL,
     [CreatedOn] DATETIME2 NOT NULL DEFAULT current_timestamp,
+    [EmployerName] NVARCHAR(100) NOT NULL , 
+    [EmployerAccountLegalEntityId] BIGINT NOT NULL, 
+    [TrainingProviderName] NVARCHAR(100) NOT NULL, 
     CONSTRAINT PK_Registration_Id PRIMARY KEY CLUSTERED ([Id]),
```
Comment (Pauljgraham) [inline src/SFA.DAS.ApprenticeCommitments.Database/Tables/Registration.sql:12]: again do we need to add teh ProviderId here?

**Review by Pauljgraham — COMMENTED** [review]
Minor comment about having a providerId

File: src/SFA.DAS.ApprenticeCommitments.Database/Tables/Apprenticeship.sql (hunk 1 above)
Comment (benagain) [reply src/SFA.DAS.ApprenticeCommitments.Database/Tables/Apprenticeship.sql]: Good point.  I've added the Provider ID and made everything mandatory.

File: src/SFA.DAS.ApprenticeCommitments.Api.AcceptanceTests/Features/GetApprenticeship.feature [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+Scenario: When the apprenticeship exists but for a different apprentice
+	Given the apprenticship exists and it's assodicated with another apprentice
+	When we try to retrieve the apprenticeship
+	Then the result should return NotFound
+
+Scenario: When the apprenticeship exists, but not for this apprentice
+	Given the apprenticeship exists, but it's associated with another apprentice
+	When we try to retrieve the apprenticeship
+	Then the result should return NotFound
```
[the remark this answers is not in the document]
Comment (benagain) [reply src/SFA.DAS.ApprenticeCommitments.Api.AcceptanceTests/Features/GetApprenticeship.feature:21]: Yep, I've removed it again.

File: src/SFA.DAS.ApprenticeCommitments.UnitTests/Application/Commands/CreateRegistrationCommandTests/WhenValidatingCreateRegistrationCommand.cs [hunk 4]
```diff
@@ -26,6 +26,23 @@ public void When_validating_Email(string email, bool expectValid)
...
+        }
+
+        [TestCase(0, false)]
+        [TestCase(-10, false)]
+        [TestCase(10, true)]
+        public void When_validating_AccountLegalEntityId(long id, bool expectValid)
+        {
+            AssertValidationResult(request => request.EmployerAccountLegalEntityId, id, expectValid);
+        }
+
```
[the remark this answers is not in the document]
Comment (benagain) [reply src/SFA.DAS.ApprenticeCommitments.UnitTests/Application/Commands/CreateRegistrationCommandTests/WhenValidatingCreateRegistrationCommand.cs:45]: Added.

File: src/SFA.DAS.ApprenticeCommitments/Application/Commands/CreateRegistrationCommand/CreateRegistrationCommandValidator.cs [hunk 5]
```diff
@@ -8,7 +8,9 @@ public CreateRegistrationCommandValidator()
         {
             RuleFor(model => model.RegistrationId).Must(id => id != default).WithMessage("The Registration Id must be valid");
             RuleFor(model => model.ApprenticeshipId).Must(id => id > 0).WithMessage("The ApprenticeshipId must be positive");
+            RuleFor(model => model.EmployerAccountLegalEntityId).Must(id => id > 0).WithMessage("The EmployerAccountLegalEntityId must be positive");
             RuleFor(model => model.Email).NotNull().EmailAddress().WithMessage("Email must be a valid email address");
+            RuleFor(model => model.EmployerName).NotEmpty().WithMessage("The Employer Name is required");
         }
```
[the remark this answers is not in the document]
Comment (benagain) [reply src/SFA.DAS.ApprenticeCommitments/Application/Commands/CreateRegistrationCommand/CreateRegistrationCommandValidator.cs:16]: Added

**Review by Pauljgraham — APPROVED** [review]

**Pull request merged**